### PR TITLE
[WIP] Set up Google analytics

### LINKED ISSUE #62: Set up Google Analytics

### PR DESCRIPTION
**Pull request by obulat**
Fixes #62 and #64 

**Description**

Set up Google analytics for chooser. The following events are tracked:

- **User opening the page**
- **User clicking on any of the 'Help' button** 
_Information sent:_ `HelpSection.clicked.<button-title>`
- **User opening Attribution Details step**, when we can assume that the user has selected a license : 
_Information sent:_ `Stepper.licenseSelected.<short name of selected license>`

- **User selecting a license from the dropdown**
_Information sent:_ `LicenseDropdown.licenseSelected.<short name of selected license>`
- **User copying attribution details**
_Information sent:_ 
     `Attribution.copied.<Stringified JSON: license: short name of selected license, codeType: one of html/rich-text, fieldsFilled: <fieldName: true/false>>`
   
**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by akmadian** [comment]
I will also add that this PR will effectively fix #64

**Comment by akmadian** [comment]
Also, please add #64 to the "Fixes" line in your PR body

File: src/main.js [hunk 1]
```diff
@@ -21,6 +22,14 @@ Vue.use(Buefy, {
     defaultIconComponent: 'vue-fontawesome',
     defaultIconPack: 'fas'
 })
+
+Vue.use(VueAnalytics, {
+    id: 'UA-XXX-X',
```
[the remark this answers is not in the document]
Comment (obulat) [reply src/main.js]: Added.

File: src/components/Stepper.vue [hunk 2]
```diff
@@ -139,11 +139,18 @@ export default {
         updateLicense(itemId) {
             const attrs = { ...this.licenseAttributes }
             attrs[itemId] = !attrs[itemId]
-            this.$emit('input', {
+            const licenseData = {
                 shortName: this.shortLicenseName(attrs),
                 fullName: this.fullLicenseName(attrs),
                 attributionDetails: this.$props.value.attributionDetails
+            }
+            this.$ga.event({
```
[the remark this answers is not in the document]
Comment (obulat) [reply src/components/Stepper.vue]: This approach really makes tracking much cleaner.

File: src/components/SelectedLicenseCode.vue [hunk 3]
```diff
@@ -78,6 +78,28 @@ export default {
...
+                // Value is the license copied
+                eventValue: this.$props.value.shortName
+            })
+            const fieldsFilledIn = []
+            for (const detail in this.$props.value.attributionDetails) {
+                if (this.$props.value.attributionDetails[detail] !== '') {
+                    fieldsFilledIn.push(detail)
+                }
+            }
+            fieldsFilledIn.forEach((field) => {
```
[the remark this answers is not in the document]
Comment (obulat) [reply src/components/SelectedLicenseCode.vue]: This, too, makes the tracking cleaner. GA doesn't let us send object, so I'm sending JSON.stringified object.

**Comment by obulat** [comment]
Thank you, Ari, your changes made it much better :)

**Pull request merged**